File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    i = 0
    while i < len(arr) - 1:
        if arr[i] != arr[i + 1]:
            answer.append(arr[i])
        i += 1

        if i == len(arr) - 1:
            answer.append(arr[i])

    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;
import java.io.*;

public class Solution {
    public int[] solution(int []arr) {
        Deque<Integer> answer = new ArrayDeque<>();
        for(int i=0; i<arr.length; i++){
            if (answer.isEmpty()||answer.peekLast()!=arr[i]){
                answer.addLast(arr[i]);
            }
        }
        
        int [] result = new int[answer.size()];

        int size = answer.size();
        for(int j=0; j<size; j++){
            result[j] = answer.pollFirst();
        }
        return result;
    }
}